### PR TITLE
My Home: Add "Remind me" button to "Connect accounts" task

### PR DESCRIPTION
**Pull request by mmtr**
#### Changes proposed in this Pull Request

This PR adds the ability to skip the "Connect accounts" task with a "Remind me" feature that will will hide the card for the specified time.

Screen Shot 2020-04-22 at 16 40 01

Since this is a feature that will be needed by all "single tasks" we're planning to implement, and all of them share the same structure, this PR also adds a new `Task` component in order to make easier the development of future tasks.

#### Testing instructions

* Apply D42273-code and sandbox the API.
* Load the experimental Home layout by going to `/home/:site?flags=home/experimental-layout` for an established site.
* Enable the analytics debug by typing `localStorage.debug='calypso:analytics*'` in the browser console and reloading My Home.
* Make sure the "Connect accounts" task shows up.
* Make sure there is a "Remind me" dropdown displaying the "Tomorrow", "Next week" and "Never" options.
* Select one of them and quickly restore the task by undoing the action (you can only do this during the 5 seconds the notice is visible).
* Make sure the card disappears, a success notice is displayed with an "Undo" button and clicking on it restores the task back.
* Make sure stats are tracked for both actions (skip and restore):
Screen Shot 2020-04-23 at 12 53 33

* Reload the experimental Home layout.
* Make sure the task is still there.
* Dismiss the task again but don't restore it this time.
* Make sure the card disappears.
* Reload the experimental Home layout.
* Make sure the card doesn't show up.
* Simulate the selected time has passed so it's time to display again the card:
  * In D42273-code, manually update the threshold to 0 for the selected reminder option. 
  * For example, if you chose "Tomorrow", make sure the threshold for the `1d` case is 0.
```php
case '1d':
	$threshold = 0;
	break;
```
* Reload the experimental Home layout again.
* Make sure the card shows up now.

### CODE REVIEW

File: client/my-sites/customer-home/cards/tasks/single-task.jsx [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+						{ actionText }
+					</Button>
+					<Button
+						className="single-task__skip is-link"
+						ref={ skipButtonRef }
+						onClick={ () => {
+							setSkipOptionsVisible( true );
+						} }
+					>
+						{ translate( 'Remind me' ) }
```
Comment (a8ci18n) [inline client/my-sites/customer-home/cards/tasks/single-task.jsx:77]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **52** times:
`translate( 'Remind Me Later' )` *ES Score: 8*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/home-tasks-temporary-dismissal)

**Comment by kwight** [comment]
This all works great @mmtr (and thanks for the `git diff` ❤️ ). A few things:

I have to admit, I'm not really a fan of this pattern: user dismisses something, and then we may (or may not) display it to them again at some point in the future. Am I the only one that gets intense FOMO and immediately starts second-guessing my dismissal? 🙃 I guess I find it more empowering to have "dismissed" things still accessible somehow, somewhere (but I have no concrete suggestion of how to present that UX to a user). But I'm certainly happy to disagree and commit at this stage. (cc/ @danhauk ).

I did find it pretty jarring when I clicked the timeframe too – something made me really surprised when the card was immediately gone (dropdowns are for setting a value, not executing the action, is what I think was in my mind for some reason). Maybe we could add a snackbar or something?.. ("Card dismissed" -> Undo).

> The card should show up after the selected time passes, but not sure how can we test that, since that's controlled server-side and I'm not aware of any safe and/or easy way to set a different time in our sandboxes.

I set each option and then moved the `0` case value between them to verify each one was properly triggered (updated test instructions).

File: client/my-sites/customer-home/cards/tasks/connect-accounts/index.jsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+						: `/marketing/connections/${ siteSlug }?tour=marketingConnectionsTour`
+				}
+				illustration="/calypso/images/stats/tasks/social-links.svg"
+				timing={ 3 }
+				dismissalPreferenceName="home-task-connect-accounts"
+			/>
+		</>
+	);
+};
+
```
Comment (kwight) [inline client/my-sites/customer-home/cards/tasks/connect-accounts/index.jsx:40]: Could we also add card-specific stats?

**Comment by mmtr** [comment]
> user dismisses something, and then we may (or may not) display it to them again at some point in the future

At least users have the control to choose that point in the future so it's not controlled by us. Although you're right that we may not display card at that point if for example the user completed the task at that point (already set up the Publicize connections) or they are no longer eligible (Publicize module is disabled on an AT site). But in both cases that's caused by a user action, so maybe the user wouldn't have an expectation of seeing that card again after performing such actions. 

> I find it more empowering to have "dismissed" things still accessible somehow

I also like this. I remember @danhauk explored this in the past, but I'm not sure if we either discarded that approach or if it's planned for a future iteration.

> the card was immediately gone [...] Maybe we could add a snackbar or something?

That's a good idea! I'll push something implementing it today.

> dropdowns are for setting a value, not executing the action

I don't find it weird, but don't have strong opinions either. FWIW we already have this pattern in other places like in the posts/pages list page:

Screen Shot 2020-04-23 at 11 45 28

File: client/my-sites/customer-home/cards/tasks/connect-accounts/index.jsx (hunk 2 above)
Comment (mmtr) [reply client/my-sites/customer-home/cards/tasks/connect-accounts/index.jsx:40]: Covered in 92e0afc

**Comment by mmtr** [comment]
Pushed a commit that shows a success notice when a task is dismissed. The notice includes an "Undo" button for restoring the task in case it's been dismissed accidentally.

Apr-23-2020 12-55-23

File: client/my-sites/customer-home/cards/tasks/task.jsx [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+						{ actionText }
+					</Button>
+					<Button
+						className="task__skip is-link"
+						ref={ skipButtonRef }
+						onClick={ () => {
+							setSkipOptionsVisible( true );
+						} }
+					>
+						{ translate( 'Remind me' ) }
```
Comment (a8ci18n) [inline client/my-sites/customer-home/cards/tasks/task.jsx:129]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **52** times:
`translate( 'Remind Me Later' )` *ES Score: 8*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/home-tasks-temporary-dismissal)

**Pull request merged**